src/net.cpp

```diff
@@ -1010,11 +1010,12 @@ void CConnman::AcceptConnection(const ListenSocket& hListenSocket) {
     // on all platforms.  Set it again here just to be sure.
     SetSocketNoDelay(hSocket);
 
-    int bannedlevel = m_banman ? m_banman->IsBannedLevel(addr) : 0;
+    bool discouraged = m_banman && m_banman->IsDiscouraged(addr);
+    bool banned = m_banman && m_banman->IsBanned(addr);
 
     // Don't accept connections from banned peers, but if our inbound slots aren't almost full, accept
     // if the only banning reason was an automatic misbehavior ban.
-    if (!NetPermissions::HasFlag(permissionFlags, NetPermissionFlags::PF_NOBAN) && bannedlevel > ((nInbound + 1 < nMaxInbound) ? 1 : 0))
+    if (!NetPermissions::HasFlag(permissionFlags, NetPermissionFlags::PF_NOBAN) && (banned || (discouraged && nInbound + 1 < nMaxInbound)))
```

Comment:
maybe this should me two checks so the log can indicate ban/discouraged

Reply:
This has been rewritten following a suggestion by @pstratem below.